#controller params
robotis_mini:
  pid_params: &id001
    p: 10.0
    i: 0.0001
    d: 0.01
    
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

  sim_position_controller:
    type: effort_controllers/JointGroupPositionController
    joints:
        - r_shoulder_joint
        - l_shoulder_joint
        - r_biceps_joint
        - l_biceps_joint
        - r_elbow_joint
        - l_elbow_joint
        - r_hip_joint
        - l_hip_joint
        - r_thigh_joint
        - l_thigh_joint
        - r_knee_joint
        - l_knee_joint
        - r_ankle_joint
        - l_ankle_joint
        - r_foot_joint
        - l_foot_joint
    l_shoulder_joint: 
      pid: *id001
    r_shoulder_joint: 
      pid: *id001
    l_biceps_joint: 
      pid: *id001
    r_biceps_joint: 
      pid: *id001
    l_elbow_joint: 
      pid: *id001
    r_elbow_joint: 
      pid: *id001
    l_hip_joint: 
      pid: *id001
    r_hip_joint: 
      pid: *id001
    l_thigh_joint:
      pid: *id001
    r_thigh_joint: 
      pid: *id001
    l_knee_joint: 
      pid: *id001
    r_knee_joint: 
      pid: *id001
    l_ankle_joint: 
      pid: *id001
    r_ankle_joint: 
      pid: *id001
    l_foot_joint: 
      pid: *id001
    r_foot_joint: 
      pid: *id001

joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50  

hw_position_controller:
  type: position_controllers/JointGroupPositionController
  joints:
      - r_shoulder_joint
      - l_shoulder_joint
      - r_biceps_joint
      - l_biceps_joint
      - r_elbow_joint
      - l_elbow_joint
      - r_hip_joint
      - l_hip_joint
      - r_thigh_joint
      - l_thigh_joint
      - r_knee_joint
      - l_knee_joint
      - r_ankle_joint
      - l_ankle_joint
      - r_foot_joint
      - l_foot_joint

joint_traj_controller:
  type: position_controllers/JointTrajectoryController
  joints:
  - r_shoulder_joint
  - l_shoulder_joint
  - r_biceps_joint
  - l_biceps_joint
  - r_elbow_joint
  - l_elbow_joint
  - r_hip_joint
  - l_hip_joint
  - r_thigh_joint
  - l_thigh_joint
  - r_knee_joint
  - l_knee_joint
  - r_ankle_joint
  - l_ankle_joint
  - r_foot_joint
  - l_foot_joint
  constraints:
      goal_time: 0.6
      stopped_velocity_tolerance: 0.05
      joint1: {trajectory: 0.1, goal: 0.1}
      joint2: {trajectory: 0.1, goal: 0.1}
  stop_trajectory_duration: 0.5
  state_publish_rate:  25
  action_monitor_rate: 10